### PR TITLE
[EP-249] Page Viewed (manage pledge)

### PR DESCRIPTION
**Pull request by tobitech**
# 📲 What

Implement page viewed event tracking for manage pledge screen

# 🤔 Why

This is part of phase 3 iOS Native events to send tracking data to segment

### CODE REVIEW

File: Library/Tracking/KSRAnalytics.swift [hunk 1]
```diff
@@ -1374,6 +1374,23 @@ public final class KSRAnalytics {
     )
   }
 
+  // MARK: - Manage Pledge Events
```
Comment (singhhari) [inline Library/Tracking/KSRAnalytics.swift:1377]: Can you remove this MARK and move the method up into the section called `Pledge Events`

File: Library/ViewModels/ManagePledgeViewModelTests.swift [hunk 2]
```diff
@@ -909,13 +909,13 @@ internal final class ManagePledgeViewModelTests: TestCase {
 
       self.scheduler.advance()
 
-      XCTAssertEqual([], self.dataLakeTrackingClient.events)
-      XCTAssertEqual([], self.segmentTrackingClient.events)
+      XCTAssertEqual(["Page Viewed"], self.dataLakeTrackingClient.events)
```
Comment (singhhari) [inline Library/ViewModels/ManagePledgeViewModelTests.swift:912]: Could you explain why this changed from [] to Page Viewed?

**Review by singhhari — CHANGES_REQUESTED** [review]
I think you have the wrong ticket number for this branch? Left a cpl of small comments.

File: Library/ViewModels/ManagePledgeViewModelTests.swift (hunk 2 above)
Comment (tobitech) [reply Library/ViewModels/ManagePledgeViewModelTests.swift:912]: Previously there was no event tracked on that view model up to the point where the fix button is tapped. But now that I added a page viewed event, the `segmentTrackingClient.events` isn't empty anymore at that point, so it has to be appended.

**Review by singhhari — APPROVED** [review]
lgtm!

**Pull request merged**